libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -240,6 +244,16 @@ public void onClick(View view) {
         return view;
     }
 
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        if (mLateInit) {
```

Comment:
Why is this "late init" logic necessary?

Reply:
Good question - I had to do this because when the activity is re-created (i.e. the app is brought to the foreground again) the post contents are set for the fragment in the activity's `onCreate` method, but the fragment still doesn't have its own stuff fully initialized (specifically, `AztecText content` is only initialized in`onCreateVIew`) so we need to wait until `onViewCreated` is called in order to be able to re-set the fragment contents with whatever we had.

There *might* be better ways to accomplish this but the EditPostActivity is already too complex to try a more aggressive refactor, which of course is needed but would demand a different amount of time and careful consideration for all the things it does.

Reply:
Stepping through the code I see we're calling `mEditorFragment.setContent` and `mEditorFragment.setTitle` three times when the activity is created:

1. In the activity's `onCreate()`
2. `SectionsPagerAdapter.instantiateItem()`
3. `fillContentEditorFields()` (which is called by `onEditorFragmentInitialized()`)

I *think* we can drop the first two and rely on the third one, which is called after the fragment is initialized - which means the "late init" logic can also be dropped.